nibabel/testing/__init__.py

```diff
@@ -57,6 +59,21 @@ def assert_allclose_safely(a, b, match_nans=True):
     assert_true(np.allclose(a, b))
 
 
+def isiterable(iterable):
+    try:
+        for _ in iterable:
+            pass
+    except:
+        return False
+
+    return True
```

Comment:
This will consume generators:

    >>> def f():
    ...    for i in range(5):
    ...        yield i
    ...
    >>> x = f()
    >>> isiterable(x)
    True
    >>> list(x)
    []

What about:

    def isiterable(val):
        try:
            iter(val)
        except TypeError:
            return False
        return True